## Program

<div style="text-align:center;padding:1em 0;"> <h4><a style="text-decoration:none;" href="https://www.zeitverschiebung.net/en/city/2761369"><span style="color:gray;">Current local time in</span> <br />Vienna, Austria</a></h4> <iframe src="https://www.zeitverschiebung.net/clock-widget-iframe-v2?language=en&size=small&timezone=Europe%2FVienna" width="100%" height="90" frameborder="0" seamless></iframe> </div>


<span style="color:red">&#9724;</span> Keynote speaker<br/>
<span style="color:blue">&#9724;</span> Invited speaker<br/>
<span style="color:orange">&#9724;</span> [Contributed talk](talks.md)<br/>
<span style="color:green">&#9724;</span> [Flash talk](posters.md)<br/>
<span style="color:gray">&#9724;</span> [Workshop session](workshop.md)



### **Monday, 15th February 2021**

| Time | Event | 
|---|---|
|09:40-10:00| *Workshop welcome session* |
|10:00-12:20| <span style="color:gray">&#9724;</span> Workshop session 1: RevBayes: Bayesian tree estimation<br/>**Sebastian Höhna** (Ludwig-Maximilians-Universität München) |
|12:20-13:40| *Lunch break* | 
|13:40-14:00| *Opening session* |
|           | Session 1 Chair: Rui Borges (Vetmeduni Vienna) |
|14:00-15:00| <span style="color:red">&#9724;</span> Species tree inference under the multispecies coalescent model<br/>**Ziheng Yang** (University College London)  |
|15:00-15:20| <span style="color:orange">&#9724;</span> Convergence assessment in Phylogenetics <br/> **Luiza Fabreti** (LMU Munich) |
|15:20-15:25| <span style="color:green">&#9724;</span> New World primate phylogeny and species boundaries with novel genomic data <br/>  **Mareike C. Janiak** (University of Salford) |
|15:25-15:30| <span style="color:green">&#9724;</span> Phylogenomic analyses of large-shelled cones from Cabo Verde, West Africa <br/> **Sara Rocha** (University of Vigo) |
|15:30-15:35| <span style="color:green">&#9724;</span> Phylogeny and species limits in African grass rats (Muridae: Arvicanthis) using genomic-scale data  <br/> **Veronika Bartáková** (Institute of Vertebrate Biology, Czech Academy of Sciences)|
|15:35-15:40| <span style="color:green">&#9724;</span> A novel approach combining diffusion approximation and Bayesian skyline plots for inferring demographic histories from SNP data <br/> **Ronja Jessica Billenstein** (LMU Munich) |
|15:40-16:00| *Break*|
|           | Session 2 Chair: ‪Gergely Szöllősi‬ (Eötvös University) |
|16:00-16:40| <span style="color:blue">&#9724;</span> Genome-scale detection of convergent molecular evolution <br/> **Bastien Boussau** (Université de Lyon) |
|16:40-17:00| <span style="color:orange">&#9724;</span> sMap: Evolution of independent, dependent and conditioned discrete characters in a Bayesian framework <br/>**Giorgio Bianchini** (School of Geographical Sciences, University of Bristol, Bristol UK) |
|17:00-17:20| <span style="color:orange">&#9724;</span> Generalizing Bayesian phylogenetics to infer shared evolutionary events<br/>**Jamie R. Oaks** (Auburn University) |
|17:20-17:25| <span style="color:green">&#9724;</span> Assessment of intra- and inter-outbreak diversity of Paenibacillus larvae by core- and whole-genome multilocus sequence typing (cg/wgMLST) <br/> **Bojan Papić** (University of Ljubljana Veterinary Faculty) |
|17:25-17:30| <span style="color:green">&#9724;</span> Reconstructing the evolutionary history of a rapid species radiation: the Andean Lupins <br/> **Bruno Nevado** (CE3C – University of Lisbon) |
|17:30-17:35| <span style="color:green">&#9724;</span> Patterns of genomic divergence and phylogenomic analyses in a young marine radiation <br/> **Martin Helmkampf** (Leibniz Centre for Tropical Marine Research (ZMT)) |
|17:35-17:40| <span style="color:green">&#9724;</span> A comparison of utility of target enrichment (TE) and double-digest restriction site associated (ddRAD) sequencing in systematics – a case study of a parapatric species pair of butterflies <br/> **Mukta Joshi** (University of Oulu) |
|17:40-18:00| *Break* |
|18:00-19:00| [Discussion group](discussiongroup.md): **Challenges and problems in phylogenomics**  |

<br>

### **Tuesday, 16th February 2021**

| Time| Event |
|---|---|
|10:00-12:20| <span style="color:gray">&#9724;</span> Workshop session 2: IQ-tree Polymorphism-aware phylogentic models <br/> **Dominik Schrempf** (Eötvös University) |
|12:20-14:00| *Lunch break* | 
|           | Session 1 Chair: Asger Hobolth (Aarhus University)  ‪ |
|14:00-14:40| <span style="color:blue">&#9724;</span> Polymorphism-ware phylogenetic models and their applications  <br/> **Carolin Kosiol** (University of St Andrews)  |
|14:40-15:00| <span style="color:orange">&#9724;</span> ILS-Aware Analyses of Retroelement Insertions in the Anomaly Zone <br/>  **Erin Molloy** (University of California, Los Angles) |
|15:00-15:20| <span style="color:orange">&#9724;</span> The corrected Synonymous Dinucleotide Usage: a novel framework for quantifying genomic dinucleotide representation <br/>  **Spyros Lytras** (MRC - University of Glasgow Centre for Virus Research) |
|15:20-15:25| <span style="color:green">&#9724;</span> Congruent birth-death models can be collapsed using Bayesian inference <br/> **Bjørn Tore Kopperud** (Ludwig-Maximilians-Universität München) |
|15:25-15:30| <span style="color:green">&#9724;</span> Phylogenetic relationships and evolution of Paederinae rove beetles (Staphylinidae) based on genomic and morphological data <br/>  **Alexandra Tokareva** (Museum and Institute of Zoology, Poland Academy of Sciences) |
|15:30-15:35| <span style="color:green">&#9724;</span> Phylogenomics analysis reveals new microsporidian species as the most basal member of Ordospora clade  <br/> **Nathalia R. de Albuquerque** (Federal University of Rio Grande do Sul)  |
|15:35-15:40| <span style="color:green">&#9724;</span> Molecular Evolution of Olfactory Receptors through jawless and jawed fishes <br/> **Liliana Silva** (CIIMAR/FCUP)  |
|15:40-16:00| *Break*|
|           | Session 2 Chair: Bastien Boussau (Université de Lyon) ‪ |
|16:00-17:00| <span style="color:red">&#9724;</span> Bayesian phylogenetic methods for analyzing paleontological and neontological data <br/> **Tracy Heath** (Iowa State University of Science and Technology) |
|17:00-17:20| <span style="color:orange">&#9724;</span> Inferring divergence times using a stepwise Bayesian approach<br/> **Allison Yi Hsiang** (LMU Munich) |
|17:20-17:40| <span style="color:orange">&#9724;</span> Dating Alphaproteobacteria evolution with eukaryotic fossils <br/> **Sishuo Wang** (The Chinese University of Hong Kong)  |
|17:40-18:00| *Break* |
|18:00-19:00| [Poster session](posters.md) |

<br>

### **Wednesday, 17th February 2021**

| Time| Event |
|---|---|
|10:00-12:20| <span style="color:gray">&#9724;</span> Workshop session 3: Bayesian polymorphism-aware phylogenetic inference  <br/>**Rui Borges** (Vetmeduni Vienna) and **Carolin Kosiol** (University of St Andrews) |
|12:20-14:00| *Lunch break* | 
|           | Session 1 Chair: ‪Carolin Kosiol (University of St Andrews) |
|14:00-14:40| <span style="color:blue">&#9724;</span> Mutational signatures in cancer genomics<br/>**Asger Hobolth** (Aarhus University) |
|14:40-15:00| <span style="color:orange">&#9724;</span> On the terminal branch length distribution and its application in the singleton density score <br/>**Benjamin Wölfl** (University of Vienna, Institute of Mathematics and Vienna Graduate School of Population Genetics) |
|15:00-15:20| <span style="color:orange">&#9724;</span> Placing ancient DNA sequences into reference phylogenies  <br/>**Bianca De Sanctis** (University of Cambridge)  |
|15:20-15:40| *Break*|
|15:40-16:00| <span style="color:orange">&#9724;</span> Dissecting genomic determinants of positive selection with an evolution-guided regression model<br/>**Yi-Fei Huang** (Pennsylvania State University) |
|16:00-16:20| <span style="color:orange">&#9724;</span> Ordering the events in the ancient tree of life with edge length ratio distributions <br/> **Andrew J. Roger** (Dalhousie University)  |
|16:20-16:40| *Closing session* |

